tensornetwork/backends/jax/jitted_functions.py

```diff
@@ -1,11 +1,42 @@
 import functools
-from typing import List, Any, Tuple, Callable, Sequence
+from typing import List, Any, Tuple, Callable, Sequence, Text
 import collections
 import types
 import numpy as np
 Tensor = Any
 
 
+def _iterative_classical_gram_schmidt(jax: types.ModuleType) -> Callable:
+
+  JaxPrecisionType = type(jax.lax.Precision.DEFAULT)
+  def iterative_classical_gram_schmidt(
+      vector: jax.ShapedArray,
+      krylov_vectors: jax.ShapedArray,
+      precision: JaxPrecisionType,
+      iterations: int = 2,
+      ) -> jax.ShapedArray:
+    """
+    orthogonalize `vector`  to all rows of `krylov_vectors`.
+    Args:
+      vector: Initial vector.
+      krylov_vectors: Matrix of krylov vectors, each row is treated as a
+        vector.
+      iterations: Number of iterations.
+    Returns:
+      jax.ShapedArray: The orthogonalized vector.
+    """
+    vec = vector
+    overlaps = 0
+    for _ in range(iterations):
+      ov = jax.numpy.dot(
```

Comment:
isn't this what vdot does?

Reply:
yes, thanks for catching that!

Reply:
actually, vdot only does scalar products